src/client/Microsoft.Identity.Client/AppConfig/PopAuthenticationConfiguration.cs

```diff
@@ -3,46 +3,97 @@
+        {
+            if (httpRequestMessage == null)
+                throw new ArgumentNullException(nameof(httpRequestMessage));
+
+            HttpMethod = httpRequestMessage.Method;
+            HttpHost = httpRequestMessage.RequestUri.Authority; // this includes the optional port
+            HttpPath = httpRequestMessage.RequestUri.AbsolutePath; 
+        }
+
+        /// <summary>
+        /// Creates a configuration using the default key management, and which binds only the Uri part of the HTTP request.
         /// </summary>
-        public Uri RequestUri { get; }
+        /// <remarks>
+        /// The UrlHost (u) and UrlPath (p) are used to create the signed HTTP request - see https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#page-3
+        /// </remarks>
+        public PoPAuthenticationConfiguration(Uri requestUri)
+        {
+            if (requestUri == null)
+                throw new ArgumentNullException(nameof(requestUri));
+
+            HttpHost = requestUri.Authority;
+            HttpPath = requestUri.AbsolutePath;
+        }
 
         /// <summary>
         /// The HTTP method ("GET", "POST" etc.) method that will be bound to the token. Leave null and the POP token will not be bound to the method.
+        /// Corresponds to the "m" part of the a signed HTTP request. Optional.
         /// </summary>
+        /// <remarks>
+        /// See https://datatracker.ietf.org/doc/html/draft-ietf-oauth-signed-http-request-03#section-3
+        /// </remarks>
         public HttpMethod HttpMethod { get; set; }
 
+        /// <summary>
+        /// The URL host of the protected API. The "u" part of a signed HTTP request.  This MAY include the port separated from the host by a colon in host:port format. Optional.
+        /// </summary>
+        public string HttpHost { get; set; }
+
+        /// <summary>
+        /// The "p" part of the signed HTTP request. 
+        /// </summary>
+        public string HttpPath { get; set; }
+
         /// <summary>
         /// An extensibility point that allows developers to define their own key management. 
         /// Leave <c>null</c> and MSAL will use a default implementation, which generates an RSA key pair in memory and refreshes it every 8 hours.
         /// Important note: if you want to change the key (e.g. rotate the key), you should create a new instance of this object,
         /// as MSAL.NET will keep a thumbprint of keys in memory.
         /// </summary>
         public IPoPCryptoProvider PopCryptoProvider { get; set; }
+
+        /// <summary>
+        /// If the protected resource (RP) requires use of a special nonce, they will publish it as part of the WWWAuthenticate header associated with a 401 HTTP response
+        /// or as part of the AuthorityInfo header associated with 200 response. Set it here to make it part of the Signed HTTP Request part of the POP token.
+        /// </summary>
+        public string Nonce { get; set; }
```

Comment:
One of the additional goals was to enable MSAL to parse the nonce value from the HttpResponse for the customer. Have you tried this out with this? Perhaps we can mentio this class in the documentation/comments? [https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/ee3cd9c2fe59c0c34ff3a65142bc16f017fd7b27/src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs#L105](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/ee3cd9c2fe59c0c34ff3a65142bc16f017fd7b27/src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs#L105)

Reply:
It does not, the format of the headers is not well understood. Tracking issue: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2844